pyop2/base.py

```diff
@@ -1825,19 +1767,41 @@ def __truediv__(self, other):
 
     def __iadd__(self, other):
         """Pointwise addition of fields."""
-        return self._iop(other, operator.iadd)
+        from numbers import Number
+        if other is not None:
+            if isinstance(other, Number):
+                self.data[:] += other
+            else:
+                np.add(self.data[:], other.data_ro, out=self.data[:], casting="unsafe")
```

Comment:
I don't think we should be doing this as its behaviour is quite undefined. We should only add things if their shapes match. If you look in `_iop` above you'll see that we call `_check_shape`. Perhaps we should call that here.

Reply:
Yes, that sounds much better. Thanks!